includes/Core/Conversion_Tracking/Conversion_Event_Providers/WooCommerce.php

```diff
@@ -54,14 +53,100 @@ public function is_active() {
 	}
 
 	/**
-	 * Gets the conversion event names that are tracked by this provider.
+	 * Gets the conversion event names that are tracked by this provider, and Google Analytics for WooCommerce addon when active.
 	 *
 	 * @since 1.127.0
 	 *
 	 * @return array List of event names.
 	 */
 	public function get_event_names() {
-		return array( 'add_to_cart', 'purchase' );
+		$wgai_event_names = $this->get_wgai_event_names();
+		$events_to_track  = $this->events_to_track();
+
+		return array_unique( array_merge( $events_to_track, (array) $wgai_event_names ) );
+	}
+
+	/**
+	 * Gets the conversion event names that are tracked by Google Analytics for WooCommerce provider.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of event names.
+	 */
+	protected function get_wgai_event_names() {
+		if ( ! class_exists( 'WC_Google_Analytics_Integration' ) ) {
+			return null;
```

Comment:
Let's return an empty array instead of null since we always merge it in the `get_event_names` method.
```suggestion
			return array();
```